nilearn/tests/test_extmath.py

```diff
@@ -6,13 +6,16 @@
 
 import numpy as np
 
+from sklearn.utils import check_random_state
 from nilearn._utils.extmath import fast_abs_percentile, is_spd
 
 
 def test_fast_abs_percentile():
-    data = np.arange(1, 100)
-    for p in range(10, 100, 10):
-        yield nose.tools.assert_equal, fast_abs_percentile(data, p - 1), p
+    rng = check_random_state(1)
+    data = np.arange(10, 100, 10)
+    rng.shuffle(data)
+    for p in data:
```

Comment:
@Loic: this PR is solely meant to fix an Index error. Those tests existed
independently of the index error. These comments have nothing to do with
current PR (lest, we'll end up having to modify tests in NiftiMasker, e.g)

On Wed, Dec 2, 2015 at 6:07 PM, Loïc Estève <notifications@github.com>
wrote:

> In nilearn/tests/test_extmath.py
> <https://github.com/nilearn/nilearn/pull/875#discussion_r46442948>:
>
> >  from nilearn._utils.extmath import fast_abs_percentile, is_spd
> >
> >
> >  def test_fast_abs_percentile():
> > -    data = np.arange(1, 100)
> > -    for p in range(10, 100, 10):
> > -        yield nose.tools.assert_equal, fast_abs_percentile(data, p - 1), p
> > +    rng = check_random_state(1)
> > +    data = np.arange(10, 100, 10)
> > +    rng.shuffle(data)
> > +    for p in data:
>
> I am not sure it is a great idea to have data and the percentiles we are
> checking to be identical. Basically you want to check the 10%, 20%, ...,
> 90% percentiles of data and data can be anything.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/875/files#r46442948>.
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob